test/inductor/test_torchinductor.py

```diff
@@ -135,6 +135,20 @@
 i32 = torch.int32
 
 
+class expectedFailureCodegenDynamicIfCUDAAndNotSM80OrLater:
+    def __init__(self, test_instance):
+        self.test_instance = test_instance
+
+    def __call__(self, fn):
+        def wrapper(slf, *args, **kwargs):
+            fn._expected_failure_codegen_dynamic = (
+                self.test_instance.device == "cuda" and not SM80OrLater
```

Comment:
Shouldn't this be `slf.device`. I would expect the decorator is only called once on the parent test class, and not for the device specific subclasses. Then again, somehow CI is passing so maybe not???

Reply:
You're right, that had the test passing as an accident.  I simplified it so that I hope it should work properly now, and it should be easier to understand.  But even this is not satisfactory, it depends on the sequential execution of tests.  Correct solution is probably just to duplicate the test code, decorate one copy with `onlyCpu` and the other one with `onlyCUDA` and `expectedFailureCodegenDynamic`, but only if `SM80OrLater`.